### PR TITLE
Add a 2-second timeout for connect()

### PR DESCRIPTION
**Pull request by intelliot**
If the rippled server is not accessible, connect() does not resolve/reject for a very long time. This adds a timeout that rejects if the connection is not made within 2 seconds (by default). Set the `connectionTimeout` property to the desired connection timeout, in milliseconds.

e.g. For a 5 second timeout, use `connectionTimeout: 5000`.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4116,16 +4116,6 @@ ripple-binary-codec@^0.2.4:
     lodash "^4.17.15"
     ripple-address-codec "^3.0.4"
 
-ripple-hashes@^0.3.4:
```
Comment (FredKSchott) [inline yarn.lock]: is this okay? seems unrelated to this PR

File: src/common/connection.ts [hunk 2]
```diff
@@ -283,7 +286,14 @@ class Connection extends EventEmitter {
 
   connect(): Promise<void> {
     this._clearReconnectTimer()
-    return new Promise((resolve, reject) => {
+    let connectFinished = false
+    const promise = new Promise<void>((resolve, reject) => {
+      setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (FKSRipple) [reply src/common/connection.ts]: fixed

**Comment by madshell** [comment]
It seems like the default value cannot be overwritten because it is not listed on  https://github.com/ripple/ripple-lib/blob/develop/src/common/schemas/input/api-options.json
